File: Xamarin/Xamarin.Native/Xamarin.Native.Droid/Services/NetworkTestService.cs
using Android.Graphics;
using Android.Util;
using Java.Net;
using System;
using System.Collections.Generic;
using System.IO;
using System.Net;
using System.Net.Http;
using System.Text;

namespace Xamarin.Services
{
    public delegate void ImageDownloadEventHandler(Bitmap image);

    internal class NetworkDownloadService : Java.Lang.Object
    {
        private static readonly NetworkDownloadService instance = new NetworkDownloadService();

        public event ImageDownloadEventHandler ImageDownloadCompleted;

        public static NetworkDownloadService Instance { get { return instance; } }

        private NetworkDownloadService()
        { }

        public void DownloadImage(string urlString)
        {
            try
            {
                URL url = new URL(urlString);
                Stream stream = url.OpenConnection().InputStream;
                Bitmap image = BitmapFactory.DecodeStream(stream);
                ImageDownloadCompleted?.Invoke(image);
            }
            catch (Exception e)
            {
                Log.Debug("Exception", "Image failed to download: " + e.ToString());
            }
        }
    }
}

File: Native.Android/app/src/main/java/com/pgrzmil/services/NetworkTestService.java
package com.pgrzmil.services;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.Log;

import java.io.InputStream;
import java.net.URL;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by pawel on 06.05.2016.
 */
public class NetworkTestService {
    private static NetworkTestService instance = new NetworkTestService();
    private List<NetworkTestListener> listeners = new ArrayList<>();

    public static NetworkTestService getInstance() {
        return instance;
    }

    private NetworkTestService() {
    }

    public void addListener(NetworkTestListener toAdd) {
        listeners.add(toAdd);
    }

    public void downloadImage(String urlString) {
        try {
            URL url = new URL(urlString);
            InputStream stream = url.openConnection().getInputStream();
            Bitmap image = BitmapFactory.decodeStream(stream);
            onImageDownloadCompleted(image);
        } catch (Exception e) {
            Log.e("Exception", "Image failed to download: " + e.toString());
        }
    }

    private void onImageDownloadCompleted(Bitmap result) {
        for (NetworkTestListener listener : listeners)
            listener.imageDownloadCompleted(result);
    }
}
